### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by michaelw8**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,4 +93,67 @@ You could use the block-form usage of the form-field component with components t
...
+                   updated=(action "update")
+                   disabled=locked}}
+```
+
+
+## Options
+Options is an object the can contain attributes and methods. Every attribute and method will be added to the component. This can be used to reach/supply build in abillities or add/overwrite event handlers.
+
+## htmlAttributes
+If you want to pass attributes to the actual html element that the form field renders, you can add 'htmlAttributes' to the options object. This object can contain thing like placeholder, name of maxlength. You can read the ember-paper documentation for more information.
```
Comment (Haz901) [inline README.md]:
```suggestion
If you want to pass attributes to the actual html element that the form field renders, you can add 'htmlAttributes' to the options object. This object can contain things like placeholder, name of maxlength. You can read the ember-paper documentation for more information.
```

Maybe add a link to the relevant section of the ember-paper documentation

File: README.md [hunk 2]
```diff
@@ -93,4 +93,67 @@ You could use the block-form usage of the form-field component with components t
...
+
+## Available types
+
+The following paper form field types are available.
+
+### number-field
+
+Renders a HTML-input[type="text"], wrapped by a component with attributes, event handlers and build-in methods. You can overwrite them by passing attributes/methods as options, as described above.
```
Comment (Haz901) [inline README.md]:
```suggestion
Renders an HTML input[type="text"] element, wrapped by a component with attributes, event handlers and build-in methods. You can overwrite them by passing attributes/methods as options, as described above. Possible attributes and methods for this field include:
```

`{{paper-input type="number"}}` renders the input with `type="number"` does our component not do this?

File: README.md [hunk 3]
```diff
@@ -93,4 +93,67 @@ You could use the block-form usage of the form-field component with components t
...
+
+**decimals** By how many decimals your number should be rounded. Default is 2.
+
+**keyDown(e)** By default you can use arrow-up and arrow-down to increase/decrease the number by one. It _won't_ round the number, so it's useable with floats.
+
+**formatOnChange(value)** Gets the value and formats its, before updating the object. By default it formats a number to '17 009,87'.
```
Comment (Haz901) [inline README.md]:
```suggestion
**formatOnChange(value)** Gets the value and formats its, before updating the object. By default it formats a number with a space as thousands separator and a comma as decimal separator, like so: '17 009,87'.
```

File: README.md [hunk 4]
```diff
@@ -93,4 +93,67 @@ You could use the block-form usage of the form-field component with components t
...
+}
+
+```
+
+```handlebars
+{{paper-form-field type="number-field"
+                   object="bakery"
+                   propertyPath="baguettes"
+                   options=options
+                   htmlAttributes=htmlAttributes
```
Comment (Haz901) [inline README.md]: So, this example shows the `htmlAttributes` passed directly to the component, while the text above describes adding them to the `options` attribute. Which method is correct?

File: README.md [hunk 5]
```diff
@@ -93,4 +93,67 @@ You could use the block-form usage of the form-field component with components t
...
+}
+
+htmlAttributes: {
+  minValue: 0
+  maxValue: 99
+}
+
+```
+
+```handlebars
```
Comment (cbroeren) [inline README.md]: I guess you could just do `hbs`

File: README.md [hunk 6]
```diff
@@ -93,4 +93,67 @@ You could use the block-form usage of the form-field component with components t
 
 ## Using your own components
 
-You could use your own components by implementing it in the form-field component like above in the block-form usage. 
+You could use your own components by implementing it in the form-field component like above in the block-form usage.
+
+# Paper-form-field
```
Comment (cbroeren) [inline README.md]: I would prefer to have the paper-form-fields above using your own component

File: README.md [hunk 7]
```diff
@@ -93,4 +93,67 @@ You could use the block-form usage of the form-field component with components t
...
+Paper-form-field is a variant of the default form-field, using ember-paper for Google Material Design like form-fields. The following sections descripe how the paper-form-field work and what options you can pass to it.
```
Comment (cbroeren) [inline README.md]: Split to multiple lines to make it more readable in code (mentioned the max-line length for every other file?)

File: README.md [hunk 8]
```diff
@@ -93,4 +93,67 @@ You could use the block-form usage of the form-field component with components t
...
+htmlAttributes: {
+  minValue: 0
+  maxValue: 99
+}
+
+```
+
+```handlebars
+{{paper-form-field type="number-field"
+                   object="bakery"
```
Comment (cbroeren) [inline README.md]: Object would never be a string?

File: paper-form-fields.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+**keyUp(e)** Gives you the ability to pass an enter or escape function.
+
+**enter()** You can pass a function called enter, that will be executed everytime you
+press the enter key. There's no default for this function.
+
+**escape()** You can pass a function called escape, that will be executed everytime
+you press the escape key. There's no default for this function.
+
+**format(value)** Gets the value and formats its, before updating the object.
```
Comment (Haz901) [inline paper-form-fields.md]: 'Gets the value and formats it'

**Pull request merged**